server/paymentProviders/thegivingblock/index.js

```diff
@@ -36,11 +62,11 @@ export async function login(login, password) {
   return apiRequest(`/login`, { method: 'POST', body });
 }
 
-export async function refresh(refreshToken) {
+export async function refresh(refreshToken, account) {
```

Comment:
The way you're adding `account` everywhere, without refactoring more, that doesn't looks good.

If going with this approach, you have to realize that `refreshToken` is already available in `account` and take the necessary steps to simplify the code.

That is basically valid for every function where you added the account parameter.

Reply:
Thanks. Yes, I've refactored the code to use `account` instead of passing `refreshToken`. The testing of this is a bit tricky so what I did was to go to the contribution flow, change the `accessToken` and then progress in the contribution flow to the QR code section. Then since the `accessToken` is not valid anymore it will throw the `INVALID_JWT_TOKEN` and hence do the re-login. This is also valid if we "lose" the refresh token for some reason. So if we change the `refreshToken` in the middle and make a request this could will handle that as well. Let me know what you think. 🤔